src/java.desktop/share/classes/javax/swing/ImageIcon.java

```diff
@@ -367,10 +378,17 @@ public Image getImage() {
 
     /**
      * Sets the image displayed by this icon.
+     * Setting a {@code null} image means
+     * any existing image will be removed
+     * and no image will be rendered.
      * @param image the image
      */
     public void setImage(Image image) {
         this.image = image;
+        if (image == null) {
+            this.description = null;
+            return;
+        }
```

Comment:
We must not change the description, but we should reset `width` and `height` to `-1`.
```suggestion
            width = -1;
            height = -1;
            return;
        }
```